libs/src/api/entityManager.ts

```diff
@@ -137,6 +137,84 @@ export class EntityManager extends Base {
     }
     return respValues
   }
+  /**
+   * Update a playlist using updated data contracts flow
+   **/
+  async updatePlaylist({
+    playlistId,
+    playlistName,
+    trackIds,
+    description,
+    isAlbum,
+    isPrivate,
+    coverArt,
+    logger = console
+  }: {
+    playlistId: number
+    playlistName: string
+    trackIds: number[]
+    description: string
+    isAlbum: boolean
+    isPrivate: boolean
+    coverArt: any
+    logger: any
+  }): Promise<{ blockHash: any; blockNumber: any; playlistId: number }> {
+    let respValues = {
+      blockHash: null,
+      blockNumber: null,
+      playlistId: 0
+    }
+    try {
+      const userId: number = parseInt(this.userStateManager.getCurrentUserId())
+      const action = 'Update'
+      const entityType = 'Playlist'
+      this.REQUIRES(Services.CREATOR_NODE)
+      // if (playlistName){
+
+      // }
+      let dirCID;
+      if (coverArt) {
+        const updatedPlaylistImage = await this.creatorNode.uploadImage(
+          coverArt,
+          true // square
+        )
+        dirCID = updatedPlaylistImage.dirCID
+      }
+
+      const playlist = (await this.discoveryProvider.getPlaylists(1, 0, [playlistId]))[0]
```

Comment:
could also check that the playlist owner id here matches the current user id, as an added precaution, but not necessary

Reply:
yeah i was wondering if we should include any validation in libs but i figured we'd ignore it for now

Reply:
great callout @jowlee - definitely worth adding, but can be done separately